### PR TITLE
Update subscription.rst

### PR DESCRIPTION
**Pull request by hannaparks**

### CODE REVIEW

**Review by cwarnermm — APPROVED** [review]
Thanks, @hannaparks!

**Review by jasonblais — CHANGES_REQUESTED** [review]
Thanks @hannaparks! Just one suggestion to hardcode it specifically to the nonprofit form

File: source/about/subscription.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ If you represent an open source community or project that is not hosted under a
 How to apply?
 ~~~~~~~~~~~~~
 
-To apply for the Mattermost Nonprofit License, `please complete this form <https://mattermost.typeform.com/to/gbM6xGkS>`__.
+To apply for the Mattermost Nonprofit License, `please use the dropddown and select Mattermost Nonprofit Program here <https://support.mattermost.com/hc/en-us/requests/new>`__.
```
[the remark this answers is not in the document]
Comment (hannaparks) [reply source/about/subscription.rst:40]: TIL

**Review by hannaparks — COMMENTED** [review]
lgtm

**Review by jasonblais — APPROVED** [review]

**Pull request merged**